### PR TITLE
Add notify me component to VPN notification settings

### PR DESCRIPTION
**Pull request by aitorvs**
Task/Issue URL: https://app.asana.com/0/0/1205842615842261/f

### Description
See asana

### Steps to test this PR

_Notify me_
- [x] install from this branch and enable NetP
- [x] remove app permissions for notifications
- [x] go to netp management screen -> vpn notifications
- [x] ensure the notify me component shows and works as expected
- [x] repeat test with notification permissions granted
- [x] verify notify me component does not show

### CODE REVIEW

**Comment by aitorvs** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #3786** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3786" target="_blank">Graphite</a>  👈
    * **PR #3787** <a href="https://app.graphite.dev/github/pr/duckduckgo/Android/3787" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/duckduckgo/Android/3786?utm_source=stack-comment).

File: network-protection/network-protection-impl/src/main/res/layout/activity_netp_notification_settings.xml [hunk 1]
```diff
@@ -54,6 +54,18 @@
                 android:layout_height="wrap_content"
                 android:orientation="vertical"
         >
+
+            <com.duckduckgo.mobile.android.ui.notifyme.NotifyMeView
+                    android:id="@+id/vpn_notification_settings_notify_me"
+                    android:layout_width="match_parent"
+                    android:layout_height="wrap_content"
+                    android:visibility="gone"
+                    app:primaryText="Get notified of VPN events"
```
Comment (karlenDimla) [inline network-protection/network-protection-impl/src/main/res/layout/activity_netp_notification_settings.xml:63]: NIT: Why not have this in donotranslate?

File: common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt [hunk 2]
```diff
@@ -281,6 +281,7 @@ class NotifyMeView @JvmOverloads constructor(
             if (newVisibility != previousVisibility) {
                 this.visibilityChangedListener?.onVisibilityChange(this, newVisibility == VISIBLE)
             }
+            setTag(tag, newVisibility)
```
Comment (karlenDimla) [inline common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:284]: Is this necessary?

File: network-protection/network-protection-impl/src/main/res/layout/activity_netp_notification_settings.xml (hunk 1 above)
Comment (aitorvs) [reply network-protection/network-protection-impl/src/main/res/layout/activity_netp_notification_settings.xml:63]: will do, missed it

File: common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt (hunk 2 above)
Comment (nalcalag) [reply common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:284]: `setTag()` is done in line 276 with the same parameters. Therefore, it looks duplicated now.
On the other hand, I understand your point so we could move `setTag()` where you put it, but remove the previous call so it's not duplicated. Does it make sense?

Comment (aitorvs) [reply common/common-ui/src/main/java/com/duckduckgo/mobile/android/ui/notifyme/NotifyMeView.kt:284]: @nalcalag is not duplicated, those two setTag execute at very different moments in time.
the one in 276 executes when calling `addViewTreeObserverOnGlobalLayoutListener`. 
The on in 284 executes in the listener, when global layout changes.

In other words, line 276 sets the initial state, line 284 sets it for the updates

**Review by karlenDimla — APPROVED** [review]

**Review by nalcalag — APPROVED** [review]
LGTM! 💯

**Pull request merged**